xmppserver/src/main/java/org/jivesoftware/openfire/commands/admin/group/UpdateGroup.java

```diff
@@ -17,32 +17,36 @@
 package org.jivesoftware.openfire.commands.admin.group;
 
 import org.dom4j.Element;
+import org.jivesoftware.openfire.SessionManager;
 import org.jivesoftware.openfire.commands.AdHocCommand;
 import org.jivesoftware.openfire.commands.SessionData;
 import org.jivesoftware.openfire.group.Group;
 import org.jivesoftware.openfire.group.GroupManager;
 import org.jivesoftware.openfire.group.GroupNotFoundException;
+import org.jivesoftware.util.LocaleUtils;
 import org.xmpp.forms.DataForm;
 import org.xmpp.forms.FormField;
 
+import javax.annotation.Nonnull;
 import java.util.Arrays;
 import java.util.Collections;
 import java.util.List;
+import java.util.Locale;
 
 /**
  * Command that allows to update a given group.
  *
  * @author Gaston Dombiak
- *
- * TODO Use i18n
  */
 public class UpdateGroup extends AdHocCommand {
     @Override
-    protected void addStageInformation(SessionData data, Element command) {
+    protected void addStageInformation(@Nonnull final SessionData data, Element command) {
+        final Locale preferredLocale = SessionManager.getInstance().getLocaleForSession(data.getOwner());
+
         DataForm form = new DataForm(DataForm.Type.form);
         if (data.getStage() == 0) {
-            form.setTitle("Update group configuration");
-            form.addInstruction("Fill out this form to specify the group to update.");
+            form.setTitle(LocaleUtils.getLocalizedString("commands.admin.group.updategroup.form.stage0.title", preferredLocale));
+            form.addInstruction(LocaleUtils.getLocalizedString("commands.admin.group.updategroup.form.stage0.instruction", preferredLocale));
 
             FormField field = form.addField();
```

Comment:
Should this be collapsed down to being 2 stages rather than 3?

Currently, it's:
* Send the command
* Send a form containing the name of the group
* Send a form containing the other details of the group

Maybe looking more like the REST request, where group name is unmodifiable key and the rest are new property values would be slicker and easier here?

Reply:
Wouldn't that remove the option of renaming the group?

Reply:
Good point

Reply:
Should we add that option? Currently the name of the group isn't on the final form.